modules/core/arrow-core/src/main/kotlin/arrow/core/Id.kt

```diff
@@ -33,4 +33,10 @@ data class Id<out A>(val value: A) : IdOf<A> {
 
     fun <A> just(a: A): Id<A> = Id(a)
   }
+
+  override fun equals(other: Any?): Boolean =
```

Comment:
What was the issue with the default implementation?

Reply:
Oh, right, you're comparing wrapped and unwrapped values.

Reply:
`Id` and the `K` Wrappers should all express their equality to the value they wrap or things can go wrong when comparing wrappers and wrapped values with `==`. This surfaced as I was testing the DSL

Reply:
We can probably fix this with inline classes. I expect them to be the same.

Reply:
Well, they aren't -.- Still, we should re-encode them as inlined at some point.